src/server/webpack.config.js

```diff
@@ -77,6 +78,22 @@ if (fs.existsSync(customConfigPath)) {
     logger.info(' => Loading custom webpack plugins.');
     config.plugins = config.plugins.concat(customConfig.plugins);
   }
+  
+  // load resolve.extensions
+  if (customConfig.resolve && customConfig.resolve.extensions) {
```

Comment:
I think it's safe to allow the whole resolve API. Not just extensions and resolve.
Adding `.js` and `node_modules` is a clever idea. I like it.